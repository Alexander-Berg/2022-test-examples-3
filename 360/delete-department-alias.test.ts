import DeleteDepartmentAlias from './delete-department-alias';

describe('delete_alias_v1', () => {
    it('works', async() => {
        const service = jest.fn(() => ({}));
        const { action } = new DeleteDepartmentAlias();

        const res = await action({ departmentId: 8, alias: 'alias', orgId: 100500 }, { service: () => service } as any);

        expect(res).toEqual({
            alias: 'alias',
            removed: true,
        });
        expect(service).toHaveBeenCalledWith('/v11/departments/8/aliases/alias/', {}, {
            method: 'DELETE',
            orgId: 100500,
        });
    });

    it('validates', async() => {
        const service = jest.fn(() => ({}));
        const { action } = new DeleteDepartmentAlias();

        await expect(action({ alias: '', departmentId: 7, orgId: 100500 }, { service: () => service } as any))
            .rejects.toMatchSnapshot('empty name');
        expect(service).not.toHaveBeenCalled();
    });
});
