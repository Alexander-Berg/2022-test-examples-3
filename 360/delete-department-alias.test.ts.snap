// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`delete_alias_v1 validates: empty name 1`] = `
DirectoryError {
  "code": "invalid_alias",
  "error": Object {
    "code": "invalid_alias",
    "message": "Alias must not be empty",
    "params": Object {
      "alias": "",
      "departmentId": 7,
      "orgId": 100500,
    },
    "type": "INVALID_PARAMS",
  },
  "type": "INVALID_PARAMS",
}
`;
